src/main/frontend/components/sidebar.cljs

```diff
@@ -149,11 +149,11 @@
 
    {:class "recent"}
 
-   (let [pages (->> (db/sub-key-value :recent/pages)
+   (let [pages (->> (db/get-key-value :recent/pages)
```

Comment:
No idea of why `db/sub-key-value` returns nil.
CC @tiensonqin

Reply:
Should abandon the function and replace the occurrence here?
https://github.com/logseq/logseq/blob/c6d03b67ab492259dd1403a405114c355953f486/src/main/frontend/components/repo.cljs#L113

Reply:
@andelf @cnrpman `sub-key-value` stopped working because it's not included when computing what queries need to be re-executed, I just pushed a fix at https://github.com/logseq/logseq/commit/30785c6a87411c3234d851d05477abd643f725ee#diff-62a577e3fda291ac77c107bdfae33966a03362d67cb8c15f95ae469c6bd5b78fR335.

`get-key-value` works too because the root component will be re-refreshed when the page route changes, but it's dangerous. 

The end-to-end tests in this PR are awesome!

Reply:
@tiensonqin  I've updated the PR and left the e2e tests only.